aptos-move/framework/aptos-framework/sources/genesis.move

```diff
@@ -126,6 +143,46 @@ module aptos_framework::genesis {
         aptos_coin::configure_accounts_for_test(aptos_framework, &core_resources, mint_cap);
     }
 
+    fun create_accounts(aptos_framework: &signer, accounts: vector<AccountMap>) {
+        let i = 0;
+        let num_accounts = vector::length(&accounts);
+
+        while (i < num_accounts) {
+            let account_map = vector::borrow(&accounts, i);
+            create_account(
+                aptos_framework,
+                account_map.account_address,
+                account_map.balance,
+            );
+
+            i = i + 1;
+        };
+    }
+
+    /// This creates an funds an account if it doesn't exist.
+    /// If it exists, it just returns the signer.
+    fun create_account(aptos_framework: &signer, account_address: address, balance: u64): signer {
+        if (account::exists_at(account_address)) {
+            create_signer(account_address)
+        } else {
+            let account = account::create_account(account_address);
+            coin::register<AptosCoin>(&account);
+            aptos_coin::mint(aptos_framework, account_address, balance);
+            account
```

Comment:
So in my previous implementation, I made this specific about failing if the account already exists.  This is to prevent there from being a double balance in the vector for the same account.

I think you should probably consider that here to abort if the account exists, at least for the `initial balances` creation run

Reply:
ahh... let's add deduplication checks to the functions that call it, by keeping around a vector of used addresses and validating they aren't used more than once.

My motivation is that we have many different entry points to prepare accounts and I'd like to share the code as much as possible. I'll probably tweak this once I wrap up with implementing the other two functions. So I'll keep this as a note to revisit the best way to enforce that constraint.